internal/controller/transform.go

```diff
@@ -0,0 +1,137 @@
+		QueryPort:              manifestquery.HTTPPort,
+		LogQueriesLongerThan:   manifests.Duration(manifests.OptionalToString(frontend.LogQueriesLongerThan)),
+		CompressResponses:      frontend.CompressResponses,
+		ResponseCacheConfig:    frontend.QueryRangeResponseCacheConfig,
+		RangeSplitInterval:     manifests.Duration(manifests.OptionalToString(frontend.QueryRangeSplitInterval)),
+		LabelsSplitInterval:    manifests.Duration(manifests.OptionalToString(frontend.LabelsSplitInterval)),
+		RangeMaxRetries:        frontend.QueryRangeMaxRetries,
+		LabelsMaxRetries:       frontend.LabelsMaxRetries,
+		LabelsDefaultTimeRange: manifests.Duration(manifests.OptionalToString(frontend.LabelsDefaultTimeRange)),
+	}
+}
+
+// QueryFrontendNameFromParent returns the name of the Thanos Query Frontend component.
+func QueryFrontendNameFromParent(resourceName string) string {
+	return fmt.Sprintf("thanos-query-frontend-%s", resourceName)
+}
+
+func rulerV1Alpha1ToOptions(in v1alpha1.ThanosRuler) manifestruler.Options {
+	labels := manifests.MergeLabels(in.GetLabels(), nil)
+	name := RulerNameFromParent(in.GetName())
+	opts := commonToOpts(name, in.GetNamespace(), in.Spec.Replicas, labels, in.Spec.CommonThanosFields, in.Spec.Additional)
+	return manifestruler.Options{
+		Options:            opts,
+		ObjStoreSecret:     in.Spec.ObjectStorageConfig.ToSecretKeySelector(),
+		Retention:          manifests.Duration(in.Spec.Retention),
+		AlertmanagerURL:    in.Spec.AlertmanagerURL,
+		ExternalLabels:     in.Spec.ExternalLabels,
+		AlertLabelDrop:     in.Spec.AlertLabelDrop,
+		StorageSize:        resource.MustParse(in.Spec.StorageSize),
+		EvaluationInterval: manifests.Duration(in.Spec.EvaluationInterval),
+	}
+}
+
+// RulerNameFromParent returns the name of the Thanos Ruler component.
+func RulerNameFromParent(resourceName string) string {
+	return fmt.Sprintf("thanos-ruler-%s", resourceName)
+}
+
+func storeV1Alpha1ToOptions(in v1alpha1.ThanosStore) manifestsstore.Options {
+	labels := manifests.MergeLabels(in.GetLabels(), in.Spec.Labels)
+	name := StoreNameFromParent(in.GetName())
+	opts := commonToOpts(name, in.GetNamespace(), in.Spec.ShardingStrategy.ShardReplicas, labels, in.Spec.CommonThanosFields, in.Spec.Additional)
+	return manifestsstore.Options{
+		ObjStoreSecret:           in.Spec.ObjectStorageConfig.ToSecretKeySelector(),
+		IndexCacheConfig:         in.Spec.IndexCacheConfig,
+		CachingBucketConfig:      in.Spec.CachingBucketConfig,
+		Min:                      manifests.Duration(manifests.OptionalToString(in.Spec.MinTime)),
+		Max:                      manifests.Duration(manifests.OptionalToString(in.Spec.MaxTime)),
+		IgnoreDeletionMarksDelay: manifests.Duration(in.Spec.IgnoreDeletionMarksDelay),
+		StorageSize:              resource.MustParse(string(in.Spec.StorageSize)),
+		Options:                  opts,
+	}
+}
+
+// StoreNameFromParent returns the name of the Thanos Store component.
+func StoreNameFromParent(resourceName string) string {
```

Comment:
I feel like these methods can just be one, and have two args, a const component type and a resourceName

Reply:
right, the reason I wanted to avoid that is so that we can have a consistent name that we expect that is learnable by calling this function across all packages. If we have to pass in the type and know the prefix upfront, we would be back to essentially generating it per package again.  

Does that make sense?